arch/arm/core/irq_manage.c

```diff
@@ -188,3 +188,65 @@ void _arch_isr_direct_header(void)
 }
 #endif
 
+#if defined(CONFIG_ARM_SECURE_FIRMWARE)
+/**
+ *
+ * @brief Set the target security state for the given IRQ
+ *
+ * Function sets the security state (Secure or Non-Secure) targeted
+ * by the given irq. It requires ARMv8-M MCU.
+ * It is only compiled if ARM_SECURE_FIRMWARE is defined.
+ * It should only be called while in Secure state, otherwise, a write attempt
+ * to NVIC.ITNS register is write-ignored(WI), as the ITNS register is not
+ * banked between security states and, therefore, has no Non-Secure instance.
+ *
+ * It shall assert if the operation is not performed successfully.
+ *
+ * @param irq IRQ line
+ * @param secure_state 1 if target state is Secure, 0 otherwise.
+ *
+ * @return N/A
+ */
+void irq_target_state_set(unsigned int irq, int secure_state)
+{
+	if (NVIC_GetTargetState(irq) == 0) {
+		/* Current target state is Secure. */
+		if (!secure_state) {
+			/* Set target state to Non-Secure */
+			if (NVIC_SetTargetState(irq) != 1) {
+				__ASSERT(0, "NVIC SetTargetState error");
+			}
+		}
+	} else {
+		/* Current target state is Non-Secure. */
+		if (secure_state) {
+			/* Set target to Secure */
+			if (NVIC_ClearTargetState(irq) != 0) {
+				__ASSERT(0, "NVIC SetTargetState error");
+			}
+		}
+	}
+}
+
+/**
+ *
+ * @brief Determine whether the given IRQ targets the Non-Secure state
+ *
+ * Function determines whether the given irq targets the Non-Secure state
+ * or not (i.e. targets the Secure state). It requires ARMv8-M MCU.
+ * It is only compiled if ARM_SECURE_FIRMWARE is defined.
+ * It should only be called while in Secure state, otherwise, a read attempt
+ * to NVIC.ITNS register is read-as-zero(RAZ), as the ITNS register is not
+ * banked between security states and, therefore, has no Non-Secure instance.
+ *
+ * @param irq IRQ line
+ *
+ * @return 1 if target state is Non-Secure, 0 otherwise.
+ */
+int irq_target_state_is_nonsecure(unsigned int irq)
```

Comment:
Are you planning irq_target_state_is_secure() ?
IMHO irq_target_state_is_secure() will better match irq_target_state_set() which uses true (non-zero) value to indicate secure IRQ.

Reply:
Right. I will keep this function as is, and, instead, align the above one to "match".

Reply:
We can just provide 2 functions: _is_secure() and _is_nonsecure(). If we make _is_secure() inline there will be no penalty.

Reply:
i could go either way on two functions.  The plus side is that it's extremely clear.  But I don't have strong feelings about it.

Reply:
For me a `if (!irq_target_state_is_nonsecure(MY_IRQ))` is just awkward. A `if (irq_target_state_is_secure(MY_IRQ))` is much easier to read.

Reply:
Then I will go with your suggestion and rename the functions, last argument makes sense, indeed.